packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -167,133 +248,121 @@ export default {
       addFiltersClause(queryParams, { id: { $notIn: knexSubQuery } });
     }
 
-    ctx.body = await strapi.entityService.findPage(targetedModel.uid, queryParams);
+    // We find a page of the targeted model (relation) to display in the
+    // relation select component in the CM edit view
+    const res = await strapi.entityService.findPage(
+      targetUid as Common.UID.ContentType,
+      queryParams
+    );
+
+    ctx.body = {
+      ...res,
+      results: Array.isArray(res.results) ? mapResults(res.results) : [],
+    };
   },
 
   async findExisting(ctx: any) {
-    const { userAbility } = ctx.state;
-    const { model, id, targetField } = ctx.params;
-
     await validateFindExisting(ctx.request.query);
-
-    const modelSchema = strapi.getModel(model);
-    if (!modelSchema) {
-      return ctx.badRequest("The model doesn't exist");
-    }
-
-    const attribute: any = modelSchema.attributes[targetField];
-    if (!attribute || attribute.type !== 'relation') {
-      return ctx.badRequest("This relational field doesn't exist");
+    const { id } = ctx.params;
+    const locale = ctx.request?.query?.locale || null;
+    const status = ctx.request?.query?.status || 'draft';
+
+    const validation = await this.extractAndValidateRequestInfo(ctx, status, id, locale);
+    if (!validation) {
+      // If validation of the request has failed the error has already been sent
+      // to the ctx
+      return;
     }
 
-    const isComponent = modelSchema.modelType === 'component';
-
-    if (!isComponent) {
-      const entityManager = getService('entity-manager');
-      const permissionChecker = getService('permission-checker').create({
-        userAbility,
-        model,
-      });
-
-      if (permissionChecker.cannot.read(null, targetField)) {
-        return ctx.forbidden();
-      }
-
-      const permissionQuery = await permissionChecker.sanitizedQuery.read(ctx.query);
-      // @ts-expect-error populate builder needs to be called with a UID
-      const populate = await getService('populate-builder')(model)
-        .populateFromQuery(permissionQuery)
-        .build();
-
-      const entity = await entityManager.findOne(id, model, { populate });
-
-      if (!entity) {
-        return ctx.notFound();
+    const { model } = ctx.params;
+    const {
+      targetField,
+      attribute,
+      fieldsToSelect,
+      sourceSchema: { modelType: sourceModelType },
+      targetSchema: { uid: targetUid },
+      currentEntityId,
+    } = validation;
+
+    // TODO type modelUid = ContentType
+    let modelUid: any = '';
+    let fields: Array<string> = locale ? [...fieldsToSelect, 'locale'] : fieldsToSelect;
+    let filters;
+    let sort = null;
+    let populate = null;
+    if (!isOneToAny(attribute)) {
```

Comment:
Couldn't we use the document service to load the relations if it's a document id? 🤔

Reply:
will dig a bit more to understand the usecases! maybe I am missing something
There was a lot of logic here , entities and components were coupled, and adding the document makes it a bit more complicated 🤔